osutil/mountentry_linux.go

```diff
@@ -26,6 +26,46 @@ import (
 	"syscall"
 )
 
+// This is needed to properly split options that could contain an escaped
+// ',' character.
+func SplitMountOptions(s string) []string {
+	var o []string
+
+	var cur strings.Builder
+	escaped := false
+	for _, c := range s {
+		switch c {
+		case '\\':
+			if escaped {
```

Comment:
Please link to the kernel source code parsing routine as a code comment.

Reply:
I added a link to the kernel source code above `SplitOverlayMountOptions` but see my comment in https://github.com/canonical/snapd/pull/14179#discussion_r1750424332.